components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml

```diff
@@ -3548,6 +3548,34 @@ paths:
         source: 'curl -k -X PUT -H "Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8"
           -H "Content-Type: application/json" -d @data.json "https://127.0.0.1:9443/api/am/admin/v2/system-scopes/role-aliases"'
 
+  ######################################################
+  # Roles resource APIs
+  ######################################################
+  /roles/{roleId}:
+    head:
+      tags:
+        - Roles
+      summary: Check Whether Given Role Name already Exist
+      description: |
+        Using this operation, user can check a given role name exists or not.
+      operationId: validateSystemRole
+      parameters:
+        - $ref: '#/components/parameters/roleId'
+      responses:
+        200:
+          description: OK. Requested role name exists.
+          content: {}
+        404:
+          $ref: '#/components/responses/NotFound'
+      security:
```

Comment:
Shall we define `500` status code also here, for documentation purpose.. In doc space, we need to display the all the possible response codes for this API call.

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9804/commits/926300b8438fa22c702bc5b8334f65f20277b17f